modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -174,5 +188,23 @@ def enum_zones
       logger.debug "Enumerated authoritative dns zones: #{zones}"
       zones
     end
+
+    def enum_records(zone_name, node_name, type)
+      records = []
+      response = execute "/EnumRecords #{zone_name} #{node_name}. /Type #{type}", "EnumRecords", true
+      response.each do |line|
+        line.chomp!
+        logger.debug "Extracting record from dnscmd output '#{line}'"
+        matches = /^@\s+(?<ttl>\d+)\s+(?<type>#{type})\s+(?<record>\S+)$/.match(line)
+        if matches.nil?
+          logger.debug "No DNS record found in this line"
+        else
+          logger.debug "Found record '#{matches['record']}'"
+          records << matches['record'] unless matches.nil?
```

Comment:
I got rid of the `next unless line =~ /\s#{type}\s/` I was using previously so `matches` is nil on the lines that don't contain records.

eg
```
D, [2016-07-01T17:49:46.282768 #14352] DEBUG -- : executing: c:\windows\sysnative\cmd.exe /c c:\Windows\System32\dnscmd.exe 10.160.148.11 /EnumRecords example.com host.example.com. /Type A
D, [2016-07-01T17:49:46.298368 #14352] DEBUG -- : Extracting record from dnscmd output 'Returned records:'
D, [2016-07-01T17:49:46.298368 #14352] DEBUG -- : No DNS record found in this line
D, [2016-07-01T17:49:46.298368 #14352] DEBUG -- : Extracting record from dnscmd output '@3600 A        10.161.93.43'
D, [2016-07-01T17:49:46.298368 #14352] DEBUG -- : Found record '10.161.93.43'
D, [2016-07-01T17:49:46.298368 #14352] DEBUG -- : Extracting record from dnscmd output ''
D, [2016-07-01T17:49:46.298368 #14352] DEBUG -- : No DNS record found in this line
D, [2016-07-01T17:49:46.298368 #14352] DEBUG -- : Extracting record from dnscmd output 'Command completed successfully.'
D, [2016-07-01T17:49:46.298368 #14352] DEBUG -- : No DNS record found in this line
D, [2016-07-01T17:49:46.298368 #14352] DEBUG -- : Extracting record from dnscmd output ''
D, [2016-07-01T17:49:46.298368 #14352] DEBUG -- : No DNS record found in this line
D, [2016-07-01T17:49:46.298368 #14352] DEBUG -- : Enumerated 1 A records for zone=example.com node=host.example.com records=["10.161.93.43"]
```